Modules/Loadable/Markups/MRMLDM/vtkMRMLMarkupsDisplayableManager.cxx

```diff
@@ -942,40 +953,24 @@ vtkSlicerMarkupsWidget * vtkMRMLMarkupsDisplayableManager::CreateWidget(vtkMRMLM
   vtkMRMLAbstractViewNode* viewNode = vtkMRMLAbstractViewNode::SafeDownCast(this->GetMRMLDisplayableNode());
   vtkRenderer* renderer = this->GetRenderer();
 
+  // Create a widget of the associated type if the node matches the registered nodes
   vtkSlicerMarkupsWidget* widget = nullptr;
-  if (vtkMRMLMarkupsAngleNode::SafeDownCast(markupsNode))
-    {
-    widget = vtkSlicerAngleWidget::New();
-    }
-  else if (vtkMRMLMarkupsFiducialNode::SafeDownCast(markupsNode))
-    {
-    widget = vtkSlicerPointsWidget::New();
-    }
-  else if (vtkMRMLMarkupsLineNode::SafeDownCast(markupsNode))
-    {
-    widget = vtkSlicerLineWidget::New();
-    }
-  else if (vtkMRMLMarkupsCurveNode::SafeDownCast(markupsNode))
+  for (const auto& item: this->MarkupsNodesWidgets)
     {
-    widget = vtkSlicerCurveWidget::New();
-    }
-  else if (vtkMRMLMarkupsClosedCurveNode::SafeDownCast(markupsNode))
-    {
-    widget = vtkSlicerClosedCurveWidget::New();
-    }
-  else if (vtkMRMLMarkupsPlaneNode::SafeDownCast(markupsNode))
-    {
-    widget = vtkSlicerPlaneWidget::New();
+    if (item.first == markupsNode->GetClassName())
+      {
+      widget = (item.second)->CreateInstance();
+      break;
+      }
     }
-  else
+
+  // If the widget was successfully createdc
+  if (widget)
     {
-    vtkErrorMacro("vtkMRMLMarkupsDisplayableManager::CreateWidget failed: invalid display node class " << markupsNode->GetClassName());
```

Comment:
Keep the main success scenario in the main branch and use if branches for handling errors (and return if needed) - similarly as it was done before. It makes the code easier to read.